x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/endpoint_activity_log.tsx

```diff
@@ -19,54 +28,97 @@ import {
   getActivityLogError,
   getActivityLogIterableData,
   getActivityLogRequestLoaded,
+  getLastLoadedActivityLogData,
   getActivityLogRequestLoading,
 } from '../../store/selectors';
 
+const Sentinel = styled.div`
+  height: 6px;
+  width: 100%;
+`;
+
 export const EndpointActivityLog = memo(
   ({ activityLog }: { activityLog: AsyncResourceState<Immutable<ActivityLog>> }) => {
     const activityLogLoading = useEndpointSelector(getActivityLogRequestLoading);
     const activityLogLoaded = useEndpointSelector(getActivityLogRequestLoaded);
+    const activityLastLogData = useEndpointSelector(getLastLoadedActivityLogData);
     const activityLogData = useEndpointSelector(getActivityLogIterableData);
+    const activityLogSize = activityLogData.length;
     const activityLogError = useEndpointSelector(getActivityLogError);
-    const dispatch = useDispatch<(a: EndpointAction) => void>();
-    const { page, pageSize } = useEndpointSelector(getActivityLogDataPaging);
+    const dispatch = useDispatch<(action: EndpointAction) => void>();
+    const { page, pageSize, disabled: isPagingDisabled } = useEndpointSelector(
+      getActivityLogDataPaging
+    );
+
+    const fetchMoreSentinel = useRef<HTMLInputElement | null>(null);
+    const getActivityLog = useCallback(
+      (entries: IntersectionObserverEntry[]) => {
+        const isIntersecting = entries.some((entry) => entry.intersectionRatio > 0);
+        if (isIntersecting && activityLogLoaded && !isPagingDisabled) {
+          dispatch({
+            type: 'appRequestedEndpointActivityLog',
+            payload: {
+              page: page + 1,
+              pageSize,
+            },
+          });
+        }
+      },
+      [activityLogLoaded, dispatch, isPagingDisabled, page, pageSize]
+    );
 
-    const getActivityLog = useCallback(() => {
-      dispatch({
-        type: 'appRequestedEndpointActivityLog',
-        payload: {
-          page: page + 1,
-          pageSize,
-        },
+    useEffect(() => {
```

Comment:
If it's because a deadline I'm ok with this approach but I would go for an agnostic approach. 
I mean, creating a component with the intersection observer and some props to manage different actions like this:
- loader: a loader component
- onScroll: callback when needs to scroll
- hasMore: flag that allows user to keep scrolling or not.
- children: component to scroll in

Reply:
Yes, we talked about this I recall. Wrapping this into a component is nicer. I just want to be sure everything works in this PR and then refactor it into a component as you suggest. I'll do this in a follow-up PR if that's okay with you.

Reply:
sure! :)

Reply:
here https://github.com/elastic/security-team/issues/1343